### PR TITLE
Add stronghold getter

### PR DESCRIPTION
**Pull request by abdulmth**
# Description of change

Adds a stronghold getter to enable access to the inner stronghold instance required by the identity library.
See https://github.com/iotaledger/identity.rs/pull/1157

Feel free to change the method name/documentation.

## Type of change

Choose a type of change, and delete any options that are not relevant.
- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested
Method is used in https://github.com/iotaledger/identity.rs/pull/1157

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Thoralf-M — CHANGES_REQUESTED** [review]
Please add a changelog entry here https://github.com/iotaledger/iota-sdk/blob/develop/sdk/CHANGELOG.md#added

**Review by Thoralf-M — DISMISSED** [review]

File: sdk/src/client/stronghold/mod.rs [hunk 1]
```diff
@@ -513,6 +516,11 @@ impl StrongholdAdapter {
 
         Ok(())
     }
+
+    /// Acquire the stronghold lock.
+    pub async fn get_stronghold(&self) -> MutexGuard<'_, Stronghold> {
```
Comment (thibault-martinez) [inline sdk/src/client/stronghold/mod.rs]: We don't usually put a `get_` prefix on our getters.
Also, `stronghold.get_stronghold()` may be a bit weird, I'd suggest just `inner` but let's see what the other think

Comment (abdulmth) [reply sdk/src/client/stronghold/mod.rs]: I think `inner` also works fine.

Comment (kwek20) [reply sdk/src/client/stronghold/mod.rs]: Id like to see .inner() instead as well

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**